packages/components/src/external-link/index.jsx

```diff
@@ -1,88 +1,91 @@
 import { localizeUrl } from '@automattic/i18n-utils';
 import clsx from 'clsx';
 import { translate } from 'i18n-calypso';
-import { omit } from 'lodash';
 import PropTypes from 'prop-types';
-import { Component } from 'react';
 import { ScreenReaderText, Gridicon } from '..';
 
 import './style.scss';
 
-class ExternalLink extends Component {
-	static defaultProps = {
-		iconSize: 18,
-		showIconFirst: false,
-		iconComponent: null,
-		localizeUrl: true,
-	};
+/**
+ * External link component that optionally includes an external link icon.
+ * @param {Object} props Component props
+ * @param {string} [props.className] Additional CSS class names
+ * @param {string} [props.href] URL the link points to
+ * @param {Function} [props.onClick] Click handler
+ * @param {boolean} [props.icon] Whether to show an external link icon
+ * @param {number} [props.iconSize] Size of the icon in pixels
+ * @param {string} [props.target] Link target attribute
+ * @param {boolean} [props.showIconFirst] Whether to show icon before the text
+ * @param {string} [props.iconClassName] Additional CSS class for the icon
+ * @param {import('react').ReactNode} [props.iconComponent] Custom icon component to use instead of default
+ * @param {boolean} [props.localizeUrl] Whether to localize the URL
+ * @param {import('react').ReactNode} [props.children] Link content
+ * @returns {import('react').ReactElement} External link component
+ */
+function ExternalLink( {
+	className,
+	icon = false,
+	iconSize = 18,
+	showIconFirst = false,
+	iconClassName,
+	iconComponent = null,
+	localizeUrl: shouldLocalizeUrl = true,
+	children,
+	...rest
```

Comment:
On the contrary, it was sad to see how the model messed up the rest props. It broke a few props, including `target`, `href`, and `onClick`. See 31cc3cbc98fc73e5985ff361f2fd3ef2d4929811 for fixing them.